internal/ssm/install_test.go

```diff
@@ -0,0 +1,132 @@
+	pgp := crypto.PGP()
+	key, err := pgp.KeyGeneration().
+		AddUserId("test", "test@example.com").
+		New().GenerateKey()
+	assert.NoError(t, err)
+
+	armoredPublicKey, err := key.GetArmoredPublicKey()
+	assert.NoError(t, err)
+
+	return armoredPublicKey, key
+}
+
+func generateSignature(t *testing.T, key *crypto.Key, data []byte) []byte {
+	pgp := crypto.PGP()
+	signer, err := pgp.Sign().SigningKey(key).Detached().New()
+	assert.NoError(t, err)
+	signature, err := signer.Sign(data, crypto.Bytes)
+	assert.NoError(t, err)
+	return signature
+}
+
+func TestInstall(t *testing.T) {
+	publicKey, privateKey := generateKeyPair(t)
+	_, wrongPrivateKey := generateKeyPair(t)
+
+	installerData := []byte("#!/bin/bash\necho 'test installer'")
+
+	tests := []struct {
+		name          string
+		installerData []byte
+		signature     []byte
+		statusCode    int
+		wantErr       bool
+	}{
+		{
+			name:          "successful installation",
+			installerData: installerData,
+			signature:     generateSignature(t, privateKey, installerData),
+			statusCode:    http.StatusOK,
+			wantErr:       false,
+		},
+		{
+			name:       "server error",
+			statusCode: http.StatusInternalServerError,
+			wantErr:    true,
+		},
+		{
+			name:          "empty installer",
+			installerData: []byte{},
+			statusCode:    http.StatusOK,
+			wantErr:       true,
+		},
+		{
+			name:          "invalid signature",
+			installerData: installerData,
+			signature:     generateSignature(t, privateKey, append(installerData, []byte("extra data to make signature invalid")...)),
+			statusCode:    http.StatusOK,
+			wantErr:       true,
+		},
+		{
+			name:          "wrong key signature",
+			installerData: installerData,
+			signature:     generateSignature(t, wrongPrivateKey, installerData),
+			statusCode:    http.StatusOK,
+			wantErr:       true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tmpDir := t.TempDir()
+			ssm.InstallerPath = filepath.Join(tmpDir, "ssm-setup-cli")
+
+			server := test.NewHTTPServer(t, func(w http.ResponseWriter, r *http.Request) {
+				if tt.statusCode != http.StatusOK {
+					w.WriteHeader(tt.statusCode)
+					return
+				}
+
+				if r.URL.Path == "/latest/linux_amd64/ssm-setup-cli" {
+					w.Write(tt.installerData)
+				} else if r.URL.Path == "/latest/linux_amd64/ssm-setup-cli.sig" {
+					w.Write(tt.signature)
+				}
+			})
+
+			source := ssm.NewSSMInstaller(zap.NewNop(), "test-region",
+				ssm.WithURLBuilder(func() (string, error) {
+					return server.URL + "/latest/linux_amd64/ssm-setup-cli", nil
+				}),
+				ssm.WithPublicKey(publicKey),
+			)
+
+			tr := &tracker.Tracker{Artifacts: &tracker.InstalledArtifacts{}}
+			logger := zap.NewNop()
+
+			err := ssm.Install(context.Background(), tr, source, logger)
+
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+
+			installedData, err := os.ReadFile(ssm.InstallerPath)
+			assert.NoError(t, err)
```

Comment:
could we use gomega? nothing against testify but it seems like gomega is more used in this repo?